server/tilesource/base.py

```diff
@@ -189,6 +203,249 @@ def _getRegionBounds(self, metadata, **kwargs):
+        size.  If none of those are specified, the original maximum resolution
+        is returned.
+          left, top, right, bottom, regionWidth, and reionHeight are used to
+        select part of the image to return in maximum resolution pixel
+        coordinates.  By default, these are the entire image.  Any two of the
+        three parameters per axis may be specified.  If all three are given,
+        the width or height is ignored.
+
+        :param width: maximum width in pixels.
+        :param height: maximum height in pixels.
+        :param left: the top of the region to output.
+        :param top: the top of the region to output.
+        :param right: the top of the region to output.
+        :param bottom: the top of the region to output.
+        :param regionWidth: the width of the region to output.
+        :param regionHeight: the height of the region to output.
+        :param units: either 'pixels' (default) or 'fraction'.  If pixels, the
+            left, top, right, bottom, regionWidth, and regionHeight are in
+            maximum resolution pixels.  If fraction, they are all on a scale of
+            0 to 1.
+        :param magnification: the magnification ratio.
+        :param mm_x: the horizontal size of a pixel in millimeters.
+        :param mm_y: the vertical size of a pixel in millimeters.
+        :param exact: if True, only a level that matches exactly will be
+            returned.  This is only applied if magnification, mm_x, or mm_y is
+            used.
+        :param format: a tuple of allowed formats.  Formats are members of
+            TILE_FORMAT_*.  This will avoid converting images if they are
+            in the desired output encoding (regardless of subparameters).
+            Otherwis,e TILE_FORMAT_PIL is returned.
+        :param **kwargs: optional arguments.  Some options are encoding,
+            jpegQuality, jpegSubsampling.
+        :returns: a dictionary of information needed for the tile iterator.
+                This is None if no tiles will be returned.  Otherwise, this
+                contains:
+            regionWidth, regionHeight: the total output of the iterator in
+                pixels.
+            xmin, ymin, xmax, ymax: the tiles that will be included during the
+                iteration: [xmin, xmax) and [ymin, ymax).
+            mode: either 'RGB' or 'RGBA'.  This determines the color space used
+                for tiles.
+            level: the tile level used for iteration.
+            metadata: tile source metadata (from getMetadata)
+            left, top: the coordinates within the image of the region returned
+                in the level pixel space.
+            width, height: the requested output resolution in pixels.  If this
+                is different that regionWidth and regionHeight, then the
+                original request was asking for a different scale that is being
+                delivered.
```

Comment:
`regionWidth` and `regionHeight` are confusing. Can you explain what are they for or how they differ from `width` and `height`?